src/Bart/Diesel.php

```diff
@@ -44,6 +51,46 @@ public static function create()
 	}
 
 	/**
+	 * Return true when PHPUnit test generator is loaded.
+	 * See comments in __callStatic method for more details.
+	 *
+	 * @return bool
+	 */
+	private static function isTestGeneratorRun()
+	{
+		if (self::$isTestGeneratorRun !== null) {
+			// Cache the result of this run time check
+			// to improve performance since resolve is called many times
+			// at run time.
+			return self::$isTestGeneratorRun;
+		}
+
+		// The test generator should have loaded the class and no autoloader should be needed
+		// when the class exists.
+		self::$isTestGeneratorRun = class_exists('\Box\TestScribe\App', false);
+
+		return self::$isTestGeneratorRun;
+	}
+
+	/**
+	 * Replace real instance with the given instance typically a mock object.
+	 *
+	 * @param string $className
+	 * @param object $instance
+	 *
+	 * @return void
+	 */
+	public static function overwrite($className, $instance)
```

Comment:
The comment should also point out when to use this versus `registerInstantiator()`